enterprise/backend/test/metabase_enterprise/advanced_permissions/api/subscription_test.clj

```diff
@@ -0,0 +1,113 @@
+(ns metabase-enterprise.advanced-permissions.api.subscription-test
+  "Permisisons tests for API that needs to be enforced by General Permissions to create and edit alerts/subscriptions."
+  (:require [clojure.test :refer :all]
+            [metabase.models :refer [Card Collection Pulse]]
+            [metabase.models.permissions :as perms]
+            [metabase.models.permissions-group :as group]
+            [metabase.public-settings.premium-features-test :as premium-features-test]
+            [metabase.test :as mt]))
+
+(defmacro ^:private temporarily-disable-subscription-for-all-users
```

Comment:
nit: it's not a big deal, but testing macros often have the `with...` naming format. you could call this something like `with-subscription-disabled-for-all-users` for consistency